cdap-ui/app/cdap/components/Alert/index.js

```diff
@@ -12,60 +12,85 @@
  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  * License for the specific language governing permissions and limitations under
  * the License.
- */
+*/
 
 import PropTypes from 'prop-types';
-
 import React, { Component } from 'react';
 import {Modal} from 'reactstrap';
+import IconSVG from 'components/IconSVG';
+
 require('./Alert.scss');
 const SUCCESS_CLOSE_TIMEOUT = 3000;
 
 export default class Alert extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showAlert: false || props.showAlert,
-      message: props.message,
-      type: props.type
-    };
-  }
+  state = {
+    showAlert: false || this.props.showAlert,
+    message: this.props.message,
+    element: this.props.element,
+    type: this.props.type
+  };
+
+  static propTypes = {
+    showAlert: PropTypes.bool,
+    message: PropTypes.string,
+    element: PropTypes.element,
+    onClose: PropTypes.func,
+    type: PropTypes.oneOf([
+      'success',
+      'error',
+      'info'
+    ])
+  };
+
   successTimeout = null;
+
   componentDidMount() {
     if (this.state.type === 'success') {
       clearTimeout(this.successTimeout);
       this.successTimeout = setTimeout(this.onClose, SUCCESS_CLOSE_TIMEOUT);
     }
   }
+
   componentWillReceiveProps(nextProps) {
-    let {showAlert, type, message} = nextProps;
+    let {showAlert, type, message, element} = nextProps;
     if (
       showAlert !== this.state.showAlert ||
       type !== this.state.type ||
-      message !== this.state.message
+      message !== this.state.message ||
+      element !== this.state.element
```

Comment:
Is this necessary? This will always be true right? `element` will be an object (react representation of dom).